dbms/src/Interpreters/Aggregator.cpp

```diff
@@ -701,49 +793,96 @@ ALWAYS_INLINE void Aggregator::executeImplBatch(
+                agg_process_info.not_found_rows.push_back(i);                                               \
+        }                                                                                                   \
+        else                                                                                                \
+        {                                                                                                   \
+            emplace_result_hold.value().setMapped(place);                                                   \
+        }                                                                                                   \
+        processed_rows = i;                                                                                 \
+    }                                                                                                       \
+    else                                                                                                    \
+    {                                                                                                       \
+        LOG_INFO(log, "HashTable resize throw ResizeException since the data is already marked for spill"); \
+        break;                                                                                              \
+    }
+
+        std::optional<size_t> processed_rows;
+        const auto end = agg_process_info.start_row + rows;
+        for (size_t i = agg_process_info.start_row; i < end; ++i)
         {
-            auto emplace_result_hold = emplaceOrFindKey<only_lookup>(
-                method,
-                state,
-                agg_process_info.start_row,
-                *aggregates_pool,
-                sort_key_containers);
-            if likely (emplace_result_hold.has_value())
+            if constexpr (enable_prefetch)
             {
-                if constexpr (collect_hit_rate)
-                {
-                    ++agg_process_info.hit_row_cnt;
-                }
-
-                if constexpr (only_lookup)
-                {
-                    if (!emplace_result_hold.value().isFound())
-                        agg_process_info.not_found_rows.push_back(i);
-                }
-                else
-                {
-                    emplace_result_hold.value().setMapped(place);
-                }
-                ++agg_process_info.start_row;
+                auto [key_holder, hashval] = getCurrentHashAndDoPrefetch(
+                    i,
+                    end,
+                    method,
+                    state,
+                    aggregates_pool,
+                    sort_key_containers,
+                    hashvals,
+                    key_holders);
+
+                auto emplace_result_hold = emplaceOrFindKey<only_lookup>(method, state, std::move(key_holder), hashval);
+
+                HANDLE_AGG_EMPLACE_RESULT
```

Comment:
It seems that this micro is unnecessary. The code of this micro can be placed after the line 880